app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java

```diff
@@ -270,21 +271,25 @@ public void onDetach() {
     void addIngredientsImage() {
         if (imagePath != null) {
             // ingredients image is already added. Open full screen image.
-            Intent intent = new Intent(getActivity(), FullScreenImage.class);
+            Intent intent = new Intent(getActivity(), FullScreenImageRotate.class);
             Bundle bundle = new Bundle();
             if (edit_product && !newImageSelected) {
                 bundle.putString("imageurl", imagePath);
+                bundle.putString("code", product.getCode());
+                bundle.putString("id", "ingredients_en");
             } else {
                 bundle.putString("imageurl", "file://" + imagePath);
+                bundle.putString("code", product.getCode());
+                bundle.putString("id", "ingredients_en");
```

Comment:
Please remove redundant code, and change it to something like:
`
            Intent intent = new Intent(getActivity(), FullScreenImageRotate.class);
            Bundle bundle = new Bundle();
            if (edit_product && !newImageSelected) {
                bundle.putString("imageurl", imagePath);
            } else {
                bundle.putString("imageurl", "file://" + imagePath);
            }
            bundle.putString("code", product.getCode());
            bundle.putString("id", "ingredients_en");
`